### PR TITLE
Disconnect from the current model when destroying it

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3558.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- connect to an existing model
- open your profile and destroy the current model
- check that you are now disconnected (it may take a few moments to disconnect)

**Comment by huwshimi** [comment]
[Disconnect when destroying the current model](https://trello.com/c/DVCi9WWW/361-disconnect-when-destroying-the-current-model)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8546/

**Comment by huwshimi** [comment]
[Re-enable the deploy button if deploying fails](https://trello.com/c/R54lq6yG/362-re-enable-the-deploy-button-if-deploying-fails)

File: jujugui/static/gui/src/app/init/utils.js [hunk 1]
```diff
@@ -344,6 +344,25 @@ utils._destroyServiceCallback = (service, db, callback, evt) => {
...
+  @param modelAPI {Object} The model API.
+  @param switchModel {Function} The method for switching models.
+  @param modelUUID {String} The UUID for the model to destroy.
+  @param callback {Function} The function to call after destroying the model.
+  @param {Boolean} clearProfileState Whether to close the profile if the model
+    is destroyed.
+*/
+utils.destroyModel = (
+  destroyModels, modelAPI, switchModel, modelUUID, callback=null, clearProfileState=true) => {
+  // If the current model is being destroyed the disconnect first.
```
Comment (hatched) [inline jujugui/static/gui/src/app/init/utils.js]: "destroyed, disconnect first."

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8548/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**